### PR TITLE
add kms decryption

### PR DESCRIPTION
**Pull request by MancunianSam**
- Decrypt environment variables with kms
- Pass the endpoint to the kms client

### CODE REVIEW

**Review by TomJKing — COMMENTED** [review]
Looks good. Just some minor comments

File: src/test/resources/application.conf [hunk 1]
```diff
@@ -5,14 +5,22 @@ s3 {
 sqs {
     endpoint = "http://localhost:8001"
     queue {
-        input = "http://localhost:8001/1/testqueueinput"
-        output = "http://localhost:8001/1/testqueueoutput"
+        input = "aHR0cDovL2xvY2FsaG9zdDo4MDAxLzEvdGVzdHF1ZXVlaW5wdXQ=" # http://localhost:8001/1/testqueueinput
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/test/resources/application.conf:8]: Makes sense, I've added it in.

File: src/test/resources/application.conf [hunk 2]
```diff
@@ -5,14 +5,22 @@ s3 {
...
     root {
-        location = "./src/test/resources/testfiles"
+        location = "Li9zcmMvdGVzdC9yZXNvdXJjZXMvdGVzdGZpbGVz" # ./src/test/resources/testfiles
     }
 }
 
-command = "test.sh result_one_record.csv"
+command = "dGVzdC5zaCByZXN1bHRfb25lX3JlY29yZC5jc3Y=" # test.sh result_one_record.csv
+
+function {
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/test/resources/application.conf:20]: Ah good spot. I've changed this one to 4 spaces as everything else is 4 spaces.

File: src/test/scala/uk.gov.nationalarchives.fileformat/FFIDExtractorTest.scala [hunk 3]
```diff
@@ -50,9 +50,9 @@ class FFIDExtractorTest extends FileSpec {
   }
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/test/scala/uk.gov.nationalarchives.fileformat/FFIDExtractorTest.scala:50]: I've removed the imports. The `right` thing is deprecated but we're using [EitherValues](https://www.scalatest.org/user_guide/using_EitherValues) It's ok in the tests, but not in the main code.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**